src/Materials/effectRenderer.ts

```diff
@@ -128,68 +94,47 @@ export class EffectRenderer {
         this.engine.drawElementsType(Constants.MATERIAL_TriangleFillMode, 0, 6);
     }
 
+    private isRenderTargetTexture(texture: InternalTexture | RenderTargetTexture): texture is RenderTargetTexture  {
+        return (texture as RenderTargetTexture).renderList !== undefined;
+    }
+
     /**
      * renders one or more effects to a specified texture
-     * @param effectWrappers list of effects to renderer
-     * @param outputTexture texture to draw to, if null it will render to the screen
+     * @param effectWrapper the effect to renderer
+     * @param outputTexture texture to draw to, if null it will render to the screen.
      */
-    public render(effectWrappers: Array<EffectWrapper> | EffectWrapper, outputTexture: Nullable<Texture> = null) {
-        if (!Array.isArray(effectWrappers)) {
-            effectWrappers = [effectWrappers];
+    public render(effectWrapper: EffectWrapper, outputTexture: Nullable<InternalTexture | RenderTargetTexture> = null) {
+        // Ensure effect is ready
+        if (!effectWrapper.effect.isReady()) {
+            return ;
         }
 
-        // Ensure all effects are ready
-        for (var wrapper of effectWrappers) {
-            if (!wrapper.effect.isReady()) {
-                return;
-            }
-        }
-
-        effectWrappers.forEach((effectWrapper, i) => {
-            var renderTo = outputTexture;
-
-            // for any next effect make it's input the output of the previous effect
-            if (i !== 0) {
-                effectWrapper.effect.onBindObservable.addOnce(() => {
-                    effectWrapper.effect.setTexture("textureSampler", this._getNextFrameBuffer(false));
-                });
-            }
+        // No need here for full screen render.

```

Comment:
basically the state changes would move there to ensure they are still called in the controls library which does not use render directly.

Reply:
Ok but what value to put instead of `this._getNextFrameBuffer(false)`:
`
effectWrapper.effect.setTexture("textureSampler", this._getNextFrameBuffer(false));
`
?

Reply:
oh no sorry just the block:
```
         // No need here for full screen render.
        this.engine.depthCullingState.depthTest = false;
        this.engine.stencilState.stencilTest = false;
```

to ensure that using applyEffectWrapper would apply it instead of render as in some app I manually use applyEffectWrapper and then draw instead of the default render

Reply:
engine state changes moved